src/services/review/RepoCheckerService.ts

```diff
@@ -316,54 +410,170 @@ export default class RepoCheckerService {
         .commit("site checker logs")
         .push(),
       (error) => new SiteCheckerError(`${error}`)
-    )
+    ).map(() => errors)
   }
 
-  checkRepo(repo: string): ResultAsync<undefined, SiteCheckerError> {
+  // adapted from https://stackoverflow.com/questions/56427009/how-to-return-papa-parsed-csv-via-promise-async-await
+  generateRepoErrorsFromCsv = (repo: string): Promise<RepoError[]> => {
+    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, REPO_LOG)
+    const file = fs.createReadStream(filePath)
+    return new Promise((resolve, reject) => {
+      Papa.parse(file, {
+        header: true,
+        complete(results) {
+          // validate the csv
+          if (!results.data.every(isRepoError)) {
+            reject(new Error("Invalid CSV"))
+          }
+          resolve(results.data as RepoError[])
+        },
+        error(error) {
+          reject(error)
+        },
+      })
+    })
+  }
+
+  readRepoErrors = (
+    repo: string
+  ): ResultAsync<BrokenLinkErrorDto, SiteCheckerError> => {
+    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, REPO_LOG)
+    if (!fs.existsSync(filePath)) {
+      logger.error(`SiteCheckerError: Folder does not exist for ${repo}`)
+      return errAsync(new SiteCheckerError(`Folder does not exist for ${repo}`))
+    }
+
+    return this.isCurrentlyErrored(repo).andThen((isError) => {
+      if (isError) {
+        return errAsync(new SiteCheckerError(`Error checking repo ${repo}`))
+      }
+
+      return this.isCurrentlyLocked(repo)
+        .andThen(() => okAsync<BrokenLinkErrorDto>({ status: "loading" }))
+        .orElse(() =>
+          ResultAsync.fromPromise(
+            this.generateRepoErrorsFromCsv(repo),
+            (error) => {
+              logger.error(
+                `SiteCheckerError: Error reading csv for repo ${repo}: ${error}`
+              )
+              return new SiteCheckerError(
+                `Error reading csv for repo ${repo}: ${error}`
+              )
+            }
+          ).andThen((errors) =>
+            okAsync<BrokenLinkErrorDto>({
+              errors,
+              status: "success",
+            })
+          )
+        )
+    })
+  }
+
+  checkRepo(repo: string): ResultAsync<BrokenLinkErrorDto, SiteCheckerError> {
     /**
      * To allow for an easy running of the script, we can temporarily clone
      * the repo to the EFS
      */
-
     let repoExists = true
+    logger.info(`Link checker for ${repo} started`)
+
+    // time the process taken to check the repo
+    const start = process.hrtime()
+    return this.isCurrentlyLocked(repo)
+      .andThen(() => okAsync<BrokenLinkErrorDto>({ status: "loading" }))
+      .orElse(() =>
+        this.isCurrentlyErrored(repo).andThen((isError) => {
+          if (isError) {
+            // delete the error.log file + retry, safe operation since this is only user triggered and not automated
+            return this.deleteErrorLog(repo).andThen(() => this.checkRepo(repo))
```

Comment:
at this point why must recurse again? just delete the error log and continue as per normal?

Reply:
fair, have modified code [here](https://github.com/isomerpages/isomercms-backend/pull/1148/commits/621e0ea37b6815c81c9ca37ada6d2057d79a620e)